test/common/storagelib.py

```diff
@@ -83,13 +83,15 @@ def add_loopback_disk(self, size=50, name=None):
         dev = self.machine.execute("F=$(mktemp /var/tmp/loop.XXXX); "
                                    "truncate --size=%sMB $F; "
                                    "losetup --show %s $F" % (size, name if name else "--find")).strip()
+        backfile = self.machine.execute(f"losetup -n -O BACK-FILE -l {dev}")
         # If this device had partions in its last incarnation on this
         # machine, they might come back for unknown reasons, in a
         # non-functional state. Running partprobe will get rid of
         # them.
         self.machine.execute("partprobe '%s'" % dev)
         # right after unmounting the device is often still busy, so retry a few times
-        self.addCleanup(self.machine.execute, f"umount {dev} || true; rm $(losetup -n -O BACK-FILE -l {dev}); until losetup -d {dev}; do sleep 1; done", timeout=10)
+        # check if loopback device detaching was successful after all.
+        self.addCleanup(self.machine.execute,f"findmnt --noheadings --output=TARGET {dev} | xargs --no-run-if-empty umount; until losetup -d {dev}; do sleep 1; done; rm {backfile}""", timeout=10)
```

Comment:
[tox rightfully complains](https://github.com/cockpit-project/cockpit/actions/runs/7724827734/job/21057758573?pr=19910) about this:
```
# test/common/storagelib.py:94:46: ISC001 Implicitly concatenated string literals on one line
```

Indeed the `""` seems redundant.

Also, as I said above: Please split this into two command executes, either with a little helper function or two addCleanups calls (remember to switch the order then). This line is too long and too complex.